pyiceberg/io/pyarrow.py

```diff
@@ -1393,9 +1393,7 @@ def _task_to_record_batches(
             # https://github.com/apache/arrow/issues/41884
             # https://github.com/apache/arrow/issues/43183
             # Would be good to remove this later on
```

Comment:
should we remove this comment? seems related to the schema type inference 
https://github.com/apache/iceberg-python/commit/1b9b884e56f74c7b7d1802774317ee95d799c5f2